include/sys/fs/zfs.h

```diff
@@ -1620,6 +1620,14 @@ typedef enum zfs_ioc {
 
 #endif
 
+typedef struct zfs_rewrite_args {
+	uint64_t	off;
+	uint64_t	len;
+	uint64_t	flags;
+} zfs_rewrite_args_t;
```

Comment:
I note that Linux's `FIDEDUPERANGE` allows a list of ranges to be provided. I don't think that's interesting here (I'm not sure it's interesting there either), the majority of uses are going to be the whole object, or a single range within, but I mention just to make sure you've thought about whether or not multiple ranges in a single call is interesting.

Reply:
I can't really guess why would I want multiple ranges there.  It would have to be pretty application-specific to know what blocks to rewrite.  I've added the range specification just to allow large file to be rewritten in several calls, if wanted.  In similar calls of other file system I was able to find I don't remember multiple range either.

Reply:
Yeah, it's the only one I know of, and imo it just complicates the interface for no obvious gain. Good!